src/background.js

```diff
@@ -330,6 +327,7 @@ const rollout = {
       // Note that this does not include the trr.mode - just that the addon cannot be ran.
       browser.experiments.heuristics.sendHeuristicsPing("prefHasUserValue", results);
       browser.experiments.preferences.clearUserPref(DOH_SELF_ENABLED_PREF);
+      await rollout.setSetting(DOH_SKIP_HEURISTICS_PREF, true);
```

Comment:
The crux of the issue – adding a line to set the skipHeuristicsCheck pref during the trrModePrefHasUserValue function.